### ISSUE TITLE
Warning: unrecognized cop always exits with code 0

### ISSUE THREAD

**Issue by coorasse** [rubocop-hq/rubocop#7231]
## Expected behavior

When a warning `Warning: unrecognized cop` is raised we expect to exit with status code 1.

## Actual behavior

`rubocop` exits with code 0.

Using `rubocop --fail-level=warning` or `Lint:
  Severity: warning` option has no effect.

## Steps to reproduce the problem
```
mkdir test-rubocop
cd test-rubocop
echo "Syntax/Whatever:\n  Enabled: true" > .rubocop.yml
rubocop
echo $?
```

0 is printed, 1 should be printed because a warning is raised.

## RuboCop version

0.73.0

**Comment by Drenmi** [comment]
It makes sense that `rubocop --fail-level=warning` should include this.
